sw/host/opentitanlib/src/transport/hyperdebug/gpio.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::Result;
+
+use crate::io::gpio::{GpioPin, PinDirection};
+use crate::transport::hyperdebug::Hyperdebug;
+
+pub struct HyperdebugGpioPin {
+    console_tty: Box<std::path::Path>,
```

Comment:
The very best in my opinion, would be if each GpioPin instance would not have its own copy, but all could refer to the same shared instance.  Maybe I can change the code to have a reference counted reference to the Inner struct of the Hyperdebug struct.  This could maybe be a fine pattern to use for several backend implementations.